src/octoprint/plugins/action_command_notification/static/js/action_command_notification.js

```diff
@@ -7,11 +7,27 @@ $(function () {
         self.settings = parameters[2];
 
         self.notifications = ko.observableArray([]);
+        self.regex = ko.observable();
         self.sortDesc = ko.observable(false);
         self.sortDesc.subscribe(function () {
             self._toLocalStorage();
         });
 
+        self.onBeforeBinding = function () {
+            self.regex(
+                self.settings.settings.plugins.action_command_notification.regex()
+            );
+        };
+
+        self.regexValid = function () {
+            try {
+                new RegExp(self.regex());
+                return true;
+            } catch (e) {
+                return false;
+            }
+        };
```

Comment:
Not sure if this would benefit from being a ko.pureComputed function or not, but relative to my other comment you could change this to use `self.settings.settings.plugins.action_command_notification.regex` and get rid of the internal regex observable and eliminate the onBeforeBinding necessity.